### ISSUE TITLE
setMaxStreams not working properly.

### ISSUE THREAD

**Issue by cgascons** [DigitalStrawberry/ANE-Sounds#11]
Hi,

I have noticed that function 'setMaxStreams' doesn't work properly, I'm passing a 1 as parameter to make it only play one sound at once and it stills plays several. Am I missing something here?

Thanks!

**Comment by marpies** [comment]
Hi,

The method takes effect only on Android platforms. It was implemented to allow changing the default number of streams allowed on Android.

On non-Android platforms there will always be multiple streams allowed to be played. Do you have the problem on Android?

**Comment by cgascons** [comment]
That's correct, sorry I missed that information, I'm indeed testing it on Android and it's not working for me.

**Comment by marpies** [comment]
Ok, what devices / system versions are you having this issue on? Can you reproduce it on different devices?

**Comment by cgascons** [comment]
I'm trying it on a Samsung Galaxy S7 running Android 7.0 and a OnePlus 3T on Android 7.0 too. I haven't got the chance to try it in any other device so far.

**Comment by cgascons** [comment]
Of course if you need me to test something just let me know!

**Comment by marpies** [comment]
I can't seem to be able to reproduce the issue on any of my devices. Are you sure you are calling `setMaxStreams` before using any other ANE API?

**Comment by cgascons** [comment]
Yes, I am @marpies, I've tried it several times already and no luck so far. Just to make sure I've got the right concept, `setMaxTreams` states the amount of sounds that can be played at the same time, right?

**Comment by marpies** [comment]
> Just to make sure I've got the right concept, setMaxTreams states the amount of sounds that can be played at the same time, right?

Yes, here's what the [official Android documentation](https://developer.android.com/reference/android/media/SoundPool) says:

> The maxStreams parameter sets the maximum number of streams that can be played at a time. SoundPool tracks the number of active streams. If the maximum number of streams is exceeded, SoundPool will automatically stop a previously playing stream based first on priority and then by age within that priority.

So with this said it may be an Android bug? If you have an APK you can share where it's reproducible I'd be happy to test it out on my devices.

**Comment by cgascons** [comment]
Hi @marpies, sorry it took me long to reply. I've just downloaded again your ANE and created a project with the assets and source you guys have inside the "test" folder, and I can replicate the error 100% of the times, I realised though I can only do it on simulator which I believe is intended, there must be something wrong on my code 100% sure since when trying the exact same code on the device works fine. Sorry about that!
